Add skip link styles

p-galligan opened the pull request:
Adds the basic styles for skipLinks.

The visually-hidden portion has not been enabled in storybook. 

There was some conversation around this in regards to React. Should we enable it in the storybook handlebars here?

@HaSistrunk @helrond @bonniegee ?

p-galligan committed:
add skip link styles

Adds styles for skipLinks.

p-galligan commented:
Additionally, `position: absolute` is layout here, but it seems pretty important to skipLinks in general. Leave it in?

helrond commented:
I would have this behave like it should IRL so let's add the visually hidden piece of the puzzle; i.e. this component should only be visible when it's focused.

Regarding positioning - that's a really good question. I'd imagine that the `top` and `left` values may somewhat depend on other page elements, like if there is a nav. I would say drop the styles in as they are in DIMES and we can review from there.

HaSistrunk commented:
I also see that in Storybook there are two color options. The orange one you've included and `skip-link--blue`. I can't remember the thinking about that to help determine what blue we might use. @helrond , do you remember if there was a specific blue or context for that other color that you had in mind?

helrond commented:
I don't recall exactly, but it may have been because I wasn't sure whether the orange would meet the WCAG baseline...or I could just have been going overboard.

HaSistrunk commented:
> I don't recall exactly, but it may have been because I wasn't sure whether the orange would meet the WCAG baseline...or I could just have been going overboard.

Cool. It _doesn't_ meet the WCAG contrast baseline now. I'd vote for us to stick with one color, leave it `$flame-orange`, and add the color contrast issue to the "Known Issues" section in the documentation for this component like we have for the button, table of contents, and pagination. Since those all have color contrast fails associated with the `$flame-orange` color, it may be easier and more consistent to address them altogether.

p-galligan committed:
add layout back

Adds top/left layout info back into style.

Makes link visually-hidden in storybook.

p-galligan commented:
> > I don't recall exactly, but it may have been because I wasn't sure whether the orange would meet the WCAG baseline...or I could just have been going overboard.
> 
> Cool. It _doesn't_ meet the WCAG contrast baseline now. I'd vote for us to stick with one color, leave it `$flame-orange`, and add the color contrast issue to the "Known Issues" section in the documentation for this component like we have for the button, table of contents, and pagination. Since those all have color contrast fails associated with the `$flame-orange` color, it may be easier and more consistent to address them altogether.

Should I update the documentation to remove the references to orange and blue?

helrond commented:
Hmm yeah that's going to take a bit of tweaking, which gets into javascript and such. I can do that, unless you want to wade in a bit more? Basically you just need to return a single SkipLink component rather than an array of them, so you'll want to do something more like the SocialIcons.

p-galligan committed:
Add hover styles

Noticed there were unanticipated hover effects. Added styles for hover.

helrond committed:
convert to single skiplink

HaSistrunk commented on stories/skipLink/skipLink.handlebars:
```diff
@@ -1 +1 @@
-<a href="#main" class="skip-link {{class}}{{#if hidden}} visually-hidden{{/if}}">{{text}}</a>
+<a href="#main" class="skip-link">{{text}}</a>
```
```suggestion
<a href="#main" class="skip-link"{{#if hidden}} visually-hidden{{/if}}>{{text}}</a>
```
Add this back in to be able to use the hidden: true/false control in Storybook?

p-galligan replied:
Yeah, sorry, thought I had added that back in already! Still trying to work on changing the handlebars to what Hillel suggested.

helrond replied:
Yeah I'm not sure about this because it's possible for this component to be dropped in a web page and to work with just HTML and CSS. Adding/removing a class can only be done via JS, so I would prefer a different pattern I think.

p-galligan committed:
Update stories/skipLink/skipLink.handlebars

add visually-hidden back in

Co-authored-by: Hannah Sistrunk <hsistrunk@rockarch.org>

helrond committed:
put story in a canvas

helrond committed:
Merge branch 'issue-66' of github.com:RockefellerArchiveCenter/styles into issue-66

helrond committed:
add accomodateion for show-hide

helrond committed:
update skiplink styles

helrond commented:
OK, so I've done a bunch of things here:
- converted the main styles to a mixin which will allow us to pass in variables for the position of the skip link. This would allow you to easily create a custom class like
```
.dimes-skip-link {
  @include skip-link($top: 60px, $left: 80px)
}
```
- used the `shown` toggle to add a bunch of inline styles which are the equivalent of `show-on-focus`. This is a little messy but I've added some documentation to indicate that those inline style are not to be used in web sites.
- made use of the `visually-hidden` and `show-on-focus` mixins.
- added some more code comments.

Curious what folks think about the mixin and positioning and whether that's useful or overkill.

helrond commented:
@p-galligan I can't request your review on this PR since you created it, so this is me requesting your review via a comment.

p-galligan reviewed (commented):
Can't approve, but looks good!

HaSistrunk commented:
Yeah, I think this approach works well and it's nice to build in the flexibility to adjust the position with the mixin.

helrond committed:
remove unused import

HaSistrunk commented on stories/skipLink/SkipLink.stories.mdx:
```diff
@@ -21,18 +21,22 @@ to main content” link on every page as the first item within the `body` elemen
 ## Known issues
 
-No known issues.
+The color flame-orange does not meet WCAG contrast requirements against the current background.
```
```suggestion
Does not meet WCAG requirements for color contrast.
```

HaSistrunk commented on stories/skipLink/SkipLink.stories.mdx:
```diff
@@ -21,18 +21,22 @@ to main content” link on every page as the first item within the `body` elemen
 
-## Colors
-Skip links can be either blue or orange. Use the `skip-link--blue` and `skip-link--orange` classes.
+The skip link component is hidden by default. To see what it looks like, switch
```
```suggestion
## Default implementation
The skip link component is hidden by default. To see what it looks like, switch
```

HaSistrunk commented on stylesheets/components/_skip-link.scss:
```diff
@@ -0,0 +1,42 @@
+  &:focus {
+    @include show-on-focus;
+
+    color: inherit; /* 1 */
```
```suggestion
    color: $desert-grey; /* 1 */
```

helrond committed:
improve documentation

helrond committed:
add explicit color to skip link
